pkg/engine/armresources.go

```diff
@@ -14,6 +14,17 @@ import (
 func GenerateARMResources(cs *api.ContainerService) []interface{} {
 	var armResources []interface{}
 
+	var deploymentTelemetryEnabled bool
+	deploymentTelemetryEnabled = cs.Properties.FeatureFlags.IsFeatureEnabled("EnableTelemetry")
+	isAzureStack := cs.Properties.IsAzureStackCloud()
+
+	if deploymentTelemetryEnabled {
+		if isAzureStack {
```

Comment:
Let's add a `validateFeatureFlags()` method to the vlabs type definition to ensure that we don't enable telemetry in non-Azure Stack scenarios. See here for an example:

https://github.com/Azure/aks-engine/blob/master/pkg/api/vlabs/validate.go#L823

So, basically let's add a new method to validate feature flags generally, but for this PR the scope is just to validate the EnableTelemetry feature flag.

We can then wire in the call to this new method in this flow:

https://github.com/Azure/aks-engine/blob/master/pkg/api/vlabs/validate.go#L152

Basically, this will be a CLI convenience so that folks who enable telemetry in a non-Azure Stack configuration scenario get a fail fast error message like "telemetry is only supported in Azure Stack..."

Lemme know if that doesn't make sense!